source/credentials_provider_sts.c

```diff
@@ -533,8 +533,13 @@ static void s_start_make_request(
     provider_user_data->signing_config.signed_body_header = AWS_SBHT_NONE;
     provider_user_data->signing_config.config_type = AWS_SIGNING_CONFIG_AWS;
     provider_user_data->signing_config.credentials_provider = impl->provider;
-    aws_date_time_init_now(&provider_user_data->signing_config.date);
-    provider_user_data->signing_config.region = s_signing_region;
+    uint64_t now = UINT64_MAX;
+    if (impl->system_clock_fn(&now) != AWS_OP_SUCCESS) {
```

Comment:
is this expected change?

Reply:
Are you asking about using the mocked system clock? We can avoid making this change and just assert on the prefix of the signature in tests; however, since we already had the option, it seemed more sensible to just use the mocked clock if set to keep it consistent.

Reply:
no was just checking whether you were intending to make this change or its something that slipped in accidentally. cause it looked somewhat unrelated to your changes. i see its for mocking now